### ISSUE TITLE
Navigation: Clicking within editor selects a new selection even when we might not want to do that

### ISSUE THREAD

**Issue by vicchig** [MajeedKazemi/nova-editor#73]
Whenever a mouse click is performed within the editor window, the code searches for the nearest code selection. However, this should not always happen. For example, if you display a suggestion within a textbox for the user and they click on it to insert some code construct at their previously selected location, it should not be the case that the selection is moved to the code construct nearest to where the cursor was during the click. We need to somehow suppress these clicks. 

In short, we don't always want to refocus the current selection every time the user clicks somewhere.

**Comment by MajeedKazemi** [comment]
urgency: 3
dependency: 1

should probably have an easy fix
